packages/relay/src/lib/relay.ts

```diff
@@ -160,13 +160,15 @@ export class RelayImpl implements Relay {
       ipAddressHbarSpendingPlanRepository,
       logger.child({ name: 'hbar-rate-limit' }),
       register,
-      Hbar.fromTinybars(total),
       duration,
     );
 
     const hapiService = new HAPIService(logger, register, this.cacheService, this.eventEmitter, hbarLimitService);
 
     this.clientMain = hapiService.getMainClientInstance();
+    if (this.clientMain.operatorAccountId) {
+      hbarLimitService.setOperatorAddress(this.clientMain.operatorAccountId.toSolidityAddress());
```

Comment:
Q: why is this needed when it's set in the `hbarLimitService` constructor?

Reply:
Sorry gotta unresolve this convo for easier tracking. Would this be considered technical debt that will be removed (if the logic is done in the service class) later when the draft PR is ready?

Reply:
It's more like a workaround due to a technical debt coming from [this PR](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3021/)